src/ClientBuilder.php

```diff
@@ -354,4 +343,34 @@ private function createEventFactory(): EventFactoryInterface
 
         return new EventFactory($this->serializer, $this->representationSerializer, $this->options, $this->sdkIdentifier, $this->getSdkVersion());
     }
+
+    public function wrapHttpClientWithPlugins(HttpAsyncClient $client): PluginClient
+    {
+        return new PluginClient($client, $this->httpClientPlugins);
+    }
+
+    private function registerDefaultPlugins(): void
+    {
+        if (0 !== \count($this->httpClientPlugins)) {
```

Comment:
I would move this `if` at the start of the `wrapHttpClientWithPlugins` method and use it to call `registerDefaultPlugins`

Reply:
@Jean85 thanks for reviewing the PR; We have applied your suggestion.